src/main/java/uk/gov/pay/products/service/PaymentCreator.java

```diff
@@ -0,0 +1,120 @@
+package uk.gov.pay.products.service;
+
+import com.google.inject.Provider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.pay.products.client.publicapi.PaymentRequest;
+import uk.gov.pay.products.client.publicapi.PaymentResponse;
+import uk.gov.pay.products.client.publicapi.PublicApiResponseErrorException;
+import uk.gov.pay.products.client.publicapi.PublicApiRestClient;
+import uk.gov.pay.products.model.Payment;
+import uk.gov.pay.products.persistence.dao.PaymentDao;
+import uk.gov.pay.products.persistence.dao.ProductDao;
+import uk.gov.pay.products.persistence.entity.PaymentEntity;
+import uk.gov.pay.products.persistence.entity.ProductEntity;
+import uk.gov.pay.products.service.transaction.NonTransactionalOperation;
+import uk.gov.pay.products.service.transaction.TransactionContext;
+import uk.gov.pay.products.service.transaction.TransactionFlow;
+import uk.gov.pay.products.service.transaction.TransactionalOperation;
+import uk.gov.pay.products.util.PaymentStatus;
+
+import javax.inject.Inject;
+
+import static uk.gov.pay.products.util.RandomIdGenerator.randomUuid;
+
+public class PaymentCreator {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private final Provider<TransactionFlow> transactionFlowProvider;
+    private final ProductDao productDao;
+    private final PaymentDao paymentDao;
+    private final PublicApiRestClient publicApiRestClient;
+    private LinksDecorator linksDecorator;
```

Comment:
Any reason why this field is not `final`?

Reply:
Made the field final.

Reply:
👍